src/main/java/com/openshift/internal/restclient/model/PersistentVolume.java

```diff
@@ -0,0 +1,276 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.internal.restclient.model;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+import com.openshift.internal.restclient.model.volume.NFSVolume;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.volume.IPersistentVolume;
+import com.openshift.restclient.model.volume.VolumeType;
+import com.openshift.restclient.utils.MemoryUnit;
+
+
+public class PersistentVolume extends KubernetesResource implements IPersistentVolume {
+
+	private static final String PV_ACCESS_MODES = "spec.accessModes";
+	private static final String PV_CAPACITY = "spec.capacity.storage";
+	private static final String PV_RECLAIM_POLICY = "spec.persistentVolumeReclaimPolicy";
+	private static final String PV_NFS = "spec.nfs";
+	private static final String PV_HOST_PATH = "spec.hostPath";
+	private static final String PV_SPEC = "spec";
+	private static final String SERVER = "server";
+	private static final String PATH = "path";
+	private static final String READ_ONLY = "readOnly";
+
+	public PersistentVolume(ModelNode node, IClient client, Map<String, String[]> propertyKeys) {
+		super(node, client, propertyKeys);
+	}
+
+	/**
+	 * @param unit the designated unit. One of "Ki", "Mi", "Gi", "Ti", "Pi", "Ei".
+	 * @return 0, if conversion not possible. Otherwise the capacity in given units.
+	 * @see {@link PersistentVolume#convert(String, MemoryUnit)}
+	 * @throws IllegalArgumentException if not supported {@code unit} given.
+	 */
+	@Override
+	public long getCapacity(String unit) {
```

Comment:
Long certainly can not hold big values in exbibytes. My assumption is that by the time we'll be creating single volumes with capacity of several Ei's, OSE3 will have been deprecated. I find it to be more convenient to use a primitive type, since it fits the most cases (if not all) and is easier to work with.